src/core/crypto.c

```diff
@@ -1647,12 +1647,55 @@ QuicCryptoProcessTlsCompletion(
         }
         Connection->Stats.ResumptionSucceeded = Crypto->TlsState.SessionResumed;
 
+        CXPLAT_DBG_ASSERT(Connection->PathsCount == 1);
+        QUIC_PATH* Path = &Connection->Paths[0];
+
+        if (Path->IsActive && Connection->Settings.IsSet.EncryptionOffloadAllowed) {
+            QUIC_CID_HASH_ENTRY* SourceCid =
+                CXPLAT_CONTAINING_RECORD(Connection->SourceCids.Next, QUIC_CID_HASH_ENTRY, Link);
+            CXPLAT_QEO_CONNECTION Offloads[] = {
+                {
+                    CXPLAT_QEO_OPERATION_ADD,
+                    CXPLAT_QEO_DIRECTION_TRANSMIT,
+                    CXPLAT_QEO_DECRYPT_FAILURE_ACTION_DROP,
+                    0, // KeyPhase
+                    0, // Reserved:0
+                    CXPLAT_QEO_CIPHER_TYPE_AEAD_AES_256_GCM,
+                    Connection->Send.NextPacketNumber,
+                },
+                {
+                    CXPLAT_QEO_OPERATION_ADD,
+                    CXPLAT_QEO_DIRECTION_RECEIVE,
+                    CXPLAT_QEO_DECRYPT_FAILURE_ACTION_DROP,
+                    0, // KeyPhase
+                    0, // Reserved:0
+                    CXPLAT_QEO_CIPHER_TYPE_AEAD_AES_256_GCM,
+                    0,
```

Comment:
`Connection->Packets[QUIC_ENCRYPT_LEVEL_1_RTT].AckTracker.LargestPacketNumberAcknowledged` should be used here.

Reply:
https://github.com/microsoft/quic-offloads/blob/main/quic-encryption-offload.md#nextpacketnumber
This says zero?

Reply:
It says _generally_, but what I have above is more accurate. Eventually, we will need to refactor this code to happen for other CIDs and keys that get created long after the handshake, and passing this in will always work.